### PR TITLE
Migrate to internal test harness

### LINKED ISSUE #1909: Migrate the maven invoker ITs tests to the internal test harness
The maven plugin has 2 types of tests:

* some using the maven invoker plugin
* some using an internal test harness mechanism

The second approach gives more flexibility and should be used for all the tests. This issue is about migrating the tests using the maven invoker plugin to the internal mechanism.

### PR DESCRIPTION
**Pull request by adrianfiedler**
Migrated the 3 maven invoker tests into SetupIT class with one test case for each. Every tested pom.xml is moved to src/test/resources/projects/...

Fixes: #1909

### CODE REVIEW

File: integration-tests/maven/src/test/java/io/quarkus/maven/it/SetupIT.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package io.quarkus.maven.it;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.*;
```
Comment (cescoffier) [inline integration-tests/maven/src/test/java/io/quarkus/maven/it/SetupIT.java:5]: It should not use * imports

**Review by cescoffier — CHANGES_REQUESTED** [review]
Looks mostly good!

Is there any other test using the Invoker plugin?

File: integration-tests/maven/src/test/java/io/quarkus/maven/it/SetupIT.java (hunk 1 above)
Comment (adrianfiedler) [reply integration-tests/maven/src/test/java/io/quarkus/maven/it/SetupIT.java]: I replaced all * imports with explicit ones

**Comment by adrianfiedler** [comment]
@cescoffier no more tests were using the maven-invoker-plugin. So I removed the dependency from pom.xml

**Comment by geoand** [comment]
If you could also squash the commits, that would be great.

**Comment by adrianfiedler** [comment]
@geoand I squashed all commits into one

**Review by cescoffier — APPROVED** [review]

**Review by machi1990 — APPROVED** [review]
LGTM.

Thanks for the PR @adrianfiedler.

**Comment by adrianfiedler** [comment]
@machi1990, @cescoffier all checks are done and ready to be merged

**Comment by machi1990** [comment]
Merged, thanks @adrianfiedler for the contribution.

**Pull request merged**